includes/collections/class-post-type.php

```diff
@@ -186,4 +187,67 @@ public static function output_collection_meta_data_for_admin_scripts( $current_s
 			);
 		}
 	}
+
+	/**
+	 * Check if a collection title already exists.
+	 *
+	 * @param string $title     The title to check.
+	 * @param int    $exclude_id Post ID to exclude from the check.
+	 * @return bool True if a collection with this title exists, false otherwise.
+	 */
+	public static function title_exists( $title, $exclude_id = 0 ) {
+		$args = [
+			'post_type'      => self::get_post_type(),
+			'post_status'    => [ 'publish', 'private', 'future', 'pending', 'draft' ],
+			's'              => $title,
+			'posts_per_page' => 100,
+			'fields'         => 'ids',
+		];
+
+		if ( ! empty( $exclude_id ) ) {
+			$args['post__not_in'] = [ $exclude_id ]; // phpcs:ignore WordPressVIPMinimum.Performance.WPQueryParams.PostNotIn_post__not_in
+		}
+
+		$posts = get_posts( $args );
+
+		// Check for exact title match.
+		foreach ( $posts as $post_id ) {
+			$post = get_post( $post_id );
+			if ( $post && trim( $post->post_title ) === trim( $title ) ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Validate collection title to ensure it's unique.
+	 *
+	 * @param int   $post_id The post ID.
+	 * @param array $data Array of unslashed post data.
+	 */
+	public static function validate_collection_title( $post_id, $data ) {
+		// Only validate for collection post type.
+		if ( self::get_post_type() !== $data['post_type'] ) {
+			return $data;
+		}
+
+		// Skip validation for auto-drafts and revisions.
+		if ( 'auto-draft' === $data['post_status'] || ! empty( $data['post_parent'] ) ) {
+			return $data;
+		}
+
+		$title = trim( $data['post_title'] );
+		if ( empty( $title ) ) {
+			return $data;
+		}
+
+		$exclude_id = ! empty( $post_id ) ? $post_id : 0;
+		$title_exists = self::title_exists( $title, $exclude_id );
+
+		if ( $title_exists ) {
+			wp_die( __( 'This collection was not published because a collection with the same title already exists. Please choose a different title and try again.', 'newspack-plugin' ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
```suggestion
			wp_die( __( 'This collection could not be saved because a collection with the same title already exists. Please choose a different title and try again.', 'newspack-plugin' ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

NIT suggestion, as every save attempt fails, not just publishing.

Reply:
Ooo, good catch! This has been updated in [4ed68d0](https://github.com/Automattic/newspack-plugin/pull/4268/commits/4ed68d0c695d9ae936e59d6574cd3924d7c93ef9) 🙂